pkg/ingester/checkpoint.go

```diff
@@ -103,7 +103,7 @@ func fromWireChunks(conf *Config, wireChunks []Chunk) ([]chunkDesc, error) {
 		// Always use Unordered headblocks during replay
 		// to ensure Loki can effectively replay an unordered-friendly
 		// WAL into a new configuration that disables unordered writes.
-		hbType := chunkenc.DefaultHeadBlockFmt
+		hbType := chunkenc.UnorderedWithNonIndexedLabelsHeadBlockFmt
```

Comment:
Has this the potential to break WAL recovery when the WAL spans across multiple periods with different schemas?

Reply:
Good point. ideally I want to figure out this format from schema config. Will fix it.

Reply:
Fixed it.